ilc/common/Environment.js

```diff
@@ -0,0 +1,20 @@
+class Environment {
+    #env;
+
+    #envVariableNames = {
+        legacyPluginsDiscovery: 'LEGACY_PLUGINS_DISCOVERY',
+    };
+
+    constructor(env) {
+        this.#env = env;
+    }
+
+    isLegacyPluginsDiscoveryEnabled() {
+        return this.#env[this.#envVariableNames.legacyPluginsDiscovery] === 'false' ? false : true;
+    }
+}
+
+module.exports = {
+    Environment,
```

Comment:
What is the idea behind exporting class and instance ?

Reply:
Class export - for testing purposes
instance is binded to process.env to prevent code duplication